@vates/nbd-client/index.js

```diff
@@ -85,6 +85,11 @@ module.exports = class NbdClient {
     // to tls during the handshake
     await this.#unsecureConnect()
     await this.#handshake()
+
+    // reset internal state if we reconnected a nbd client
+    this.#commandQueryBacklog = new Map()
+    this.#nbCommands = 0
+    this.#waitingForResponse = false
```

Comment:
If you initialize this here, I believe the constructor initialization is no longer necessary.

Reply:
also nbCommands is not really needed